pkg/operator/collection.go

```diff
@@ -328,7 +319,7 @@ func setConfigMapData(cm *corev1.ConfigMap, c monitoringv1.CompressionType, key
 
 // ensureCollectorConfig generates the collector config and creates or updates it.
 func (r *collectionReconciler) ensureCollectorConfig(ctx context.Context, spec *monitoringv1.CollectionSpec, compression monitoringv1.CompressionType, exports []monitoringv1.ExportSpec) error {
-	cfg, err := r.makeCollectorConfig(ctx, spec, exports)
+	cfg, fullUpdates, statusUpdates, err := r.makeCollectorConfig(ctx, spec, exports)
```

Comment:
Nice, after reading whole PR I understand this, but this line alone was confusing (what full means, that one contain all updates, but second only status updates for same obj?).

Given whole PR context I would consider (just an idea):

* separate type for statuses
* method for ScheduleUpdate(obj CRD, patchStatusOnly bool)
* Then in this function do Update() which will hide non-trivial set of Update + patch calls.

Another benefit of the above is that I feel we risk a bit of efficiency regressions with it. Even if not critical for majority of cases, we might need to make it more complex for scalability (e.g. do async, batch it, backoff etc)